docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/FeatureSelection/SelectFeaturesBasedOnMutualInformation.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic
+{
+    public static class SelectFeaturesBasedOnMutualInformation
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable and convert it to an IDataView.
+            var rawData = GetData();
+
+            Console.WriteLine("Contents of two columns 'Label' and 'NumericVector'.");
+            foreach (var item in rawData)
+                Console.WriteLine("{0}\t\t{1}", item.Label, string.Join(" ", item.NumericVector));
+            // True            4 0 6
+            // False           0 5 7
+            // True            4 0 6
+            // False           0 5 7
+
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            // We define a MutualInformationFeatureSelectingEstimator that selects the top k slots in a feature 
+            // vector based on highest mutual information between that slot and a specified label. 
+
+            var pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnMutualInformation(
```

Comment:
>SelectFeaturesBasedOnMutualInformation [](start = 65, length = 38)

we should have a "tutorial" to show the computation of MI..something along the lines of 

[https://www.researchgate.net/post/How_can_i_calculate_Mutual_Information_theory_from_a_simple_dataset](https://www.researchgate.net/post/How_can_i_calculate_Mutual_Information_theory_from_a_simple_dataset)

this sample is only for API reference, so small in-memory dataset suffices for this.

Reply:
+1

---
In reply to: [272375725](https://github.com/dotnet/machinelearning/pull/3184#discussion_r272375725) [](ancestors = 272375725)